pkg/feature/api/flag_trigger.go

```diff
@@ -17,62 +17,1010 @@ package api
+			}
+			return nil, dt.Err()
+		}
+		dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalize(locale.InternalServerError),
+		})
+		if err != nil {
+			return nil, statusInternal.Err()
+		}
+		return nil, dt.Err()
+	}
+	flagTriggerSecret := domain.NewFlagTriggerSecret(
+		flagTrigger.GetId(),
+		request.CreateFlagTriggerCommand.FeatureId,
+		request.EnvironmentNamespace,
+		flagTrigger.GetUuid(),
+		int(request.CreateFlagTriggerCommand.Action),
+	)
+	triggerURL, err := s.generateTriggerURL(ctx, flagTriggerSecret, false)
+	if err != nil {
+		s.logger.Error(
+			"Failed to generate trigger url",
+			log.FieldsFromImcomingContext(ctx).AddFields(zap.Error(err))...,
+		)
+		return nil, err
+	}
+	return &featureproto.CreateFlagTriggerResponse{
+		FlagTrigger: flagTrigger.FlagTrigger,
+		Url:         triggerURL,
+	}, nil
 }
 
 func (s *FeatureService) UpdateFlagTrigger(
 	ctx context.Context,
 	request *featureproto.UpdateFlagTriggerRequest,
 ) (*featureproto.UpdateFlagTriggerResponse, error) {
+	localizer := locale.NewLocalizer(ctx)
+	editor, err := s.checkRole(ctx, accountproto.Account_EDITOR, request.EnvironmentNamespace, localizer)
+	if err != nil {
+		return nil, err
+	}
+	if err := validateUpdateFlagTriggerCommand(request.ChangeFlagTriggerDescriptionCommand, localizer); err != nil {
+		s.logger.Info(
+			"Invalid argument",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+				zap.String("environmentNamespace", request.EnvironmentNamespace),
+			)...,
+		)
+		return nil, err
+	}
+	tx, err := s.mysqlClient.BeginTx(ctx)
+	if err != nil {
+		s.logger.Error(
+			"Failed to begin transaction",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+				zap.String("environmentNamespace", request.EnvironmentNamespace),
+			)...,
+		)
+	}
+	err = s.mysqlClient.RunInTransaction(ctx, tx, func() error {
+		storage := v2fs.NewFlagTriggerStorage(tx)
+		flagTrigger, err := storage.GetFlagTrigger(
+			ctx,
+			request.ChangeFlagTriggerDescriptionCommand.Id,
+			request.EnvironmentNamespace,
+		)
+		if err != nil {
+			s.logger.Error(
+				"Failed to get flag trigger",
+				log.FieldsFromImcomingContext(ctx).AddFields(
+					zap.Error(err),
+					zap.String("environmentNamespace", request.EnvironmentNamespace),
+				)...,
+			)
+			return err
+		}
+		handler := command.NewFlagTriggerCommandHandler(
+			editor,
+			flagTrigger,
+			s.domainPublisher,
+			request.EnvironmentNamespace,
+		)
+		if err := handler.Handle(ctx, request.ChangeFlagTriggerDescriptionCommand); err != nil {
+			s.logger.Error(
+				"Failed to update flag trigger",
+				log.FieldsFromImcomingContext(ctx).AddFields(zap.Error(err))...,
+			)
+			return err
+		}
+		if err := storage.UpdateFlagTrigger(
+			ctx,
+			request.ChangeFlagTriggerDescriptionCommand.Id,
+			request.EnvironmentNamespace,
+			request.ChangeFlagTriggerDescriptionCommand.Description,
```

Comment:
Please do the description change logic in the domain object via command.

### Command
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/experiment/command/experiment.go#L116-L127

### Domain
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/experiment/domain/experiment.go#L181-L185